### PR TITLE
Rework QAccessibleInterface lifetimes

### PR DESCRIPTION
**Pull request by reichlfl**
This fixes the crash reports sent from the E2E tests. Basically, whenever Accessibility info is queried from the capture window during a run of Orbit, a crash in `QAccessibleCache::deleteInterface(unsigned int,QObject *)` appears when closing down Orbit. This never shows up for the user, but spams the crash reporting.

Core issue is that we are creating instances of `QAccessibleInterface` which are not associated with a valid `QObject`, and this breaks tracking of those interfaces inside QT. To fix this, we should not create and manage `QAccessibleInterface` instances ourselfs, but instead manage dummy `QObject` instances and rely on QTs automatic creation of `QAccessibleInterface`.

Bug: b/177509546

There is another issue that is still present and not fixed with this PR, but is not exclusively related to the custom accessibility of the capture window: b/177655659

### CODE REVIEW

**Comment by reichlfl** [comment]
@beckerhe @pierricgimmig This is still a draft because I need to update the in-code and external documentation first to reflect the changes, would be great if you could have a look already and share your thoughts.

**Review by beckerhe — COMMENTED** [review]
Thank you Florian. Assuming I understand what's happening that looks good to me. I added some very minor comments. I wanna have another look when it's non-draft anymore.

File: src/OrbitQt/AccessibilityAdapter.cpp [hunk 1]
```diff
@@ -107,38 +107,25 @@
...
-    AccessibilityAdapter::all_interfaces_map_;
-absl::flat_hash_map<const AccessibleInterface*, std::unique_ptr<AccessibilityAdapter>>
-    AccessibilityAdapter::managed_adapters_;
+AdapterRegistry& AdapterRegistry::Get() {
+  static AdapterRegistry registry;
+  static std::once_flag flag;
+
+  std::call_once(flag, [&]() {
+    AccessibleInterfaceRegistry::Get().SetOnUnregisterCallback(
+        std::bind(&AdapterRegistry::OnInterfaceDeleted, &registry, std::placeholders::_1));
```
[the remark this answers is not in the document]
Comment (reichlfl) [reply src/OrbitQt/AccessibilityAdapter.cpp]: Agreed, changed to lambda

**Comment by reichlfl** [comment]
Thanks for the comments, Henning! I've addressed them and updated the documentation, both in code and at go/stadia-orbit-capture-window-e2e. PTAL.

**Review by beckerhe — APPROVED** [review]
Thanks for addressing my comments. LGTM!

**Pull request merged**